### DISCUSSION TITLE
Proxy Settings in Bottlerocket Config File

### DISCUSSION THREAD

**Discussion by ku1918** [bottlerocket-os/bottlerocket#1761] [Q&A]
Hi all,

I would like to ask what is the settings for **settings.network.https-proxy** since it is not mention in the doc.

Example:
[settings.container-registry.mirrors]
"docker.io" = ["https://<my-docker-hub-mirror-host>"]
"gcr.io" = ["https://<my-gcr-mirror-host>","http://<my-gcr-mirror-host-2>"]

So what is the config for the following settings? 

[settings.network.https-proxy]

[settings.network.no-proxy]

**Comment by etungsten** [comment]
Hi @ku1918,

`settings.network.https-proxy` accepts a single valid URL string for your proxy server (e.g. `"https://SERVER:PORT/"`).
`settings.network.no-proxy` can be list of hosts, IP addresses, or IP CIDRs you do not wish to proxy (e.g. `["example.com", ".example.com", "123.123.123.123", "10.1.0.0/16"]`). It can also accept `["*"]` which effectively disables the proxy.

You can read more about the proxy settings [here](https://github.com/bottlerocket-os/bottlerocket#proxy-settings).

Please let us know if you need further clarification!

**Comment by ku1918** [comment]
@etungsten 


So it will be like this? Because the doc doesn't tell me what is the "key".

[settings.network.https-proxy]
value = "https://SERVER:PORT/"

[settings.network.no-proxy]
value = ["example.com", ".example.com", "123.123.123.123", "10.1.0.0/16"]

**Reply by etungsten** [reply]
The key would be setting's name so either of the following:
```
[settings.network]
https-proxy = "https://SERVER:PORT/"
no-proxy = ["example.com", ".example.com", "123.123.123.123", "10.1.0.0/16"]
```
or
```
settings.network.https-proxy = "https://SERVER:PORT/"
settings.network.no-proxy = ["example.com", ".example.com", "123.123.123.123", "10.1.0.0/16"]
```

**Reply by ku1918** [reply]
ok thank you.

I will try it out.

One suggestion for the Proxy Settings documentation would be providing example like what you have mention. Maybe it could be much more clearer to first time user like me.

Thanks.
